src/os/file_nintendoswitch.go

```diff
@@ -0,0 +1,33 @@
+// +build nintendoswitch
+
+package os
+
+import "unsafe"
+
+// Read is unsupported on this system.
+func (f *File) Read(b []byte) (n int, err error) {
+	n = libc_read(int(f.fd), unsafe.Pointer(&b[0]), uint64(len(b)))
```

Comment:
Is there something about src/os/file_unix.go that doesn't work?
In general, it's better to use the syscall package for these calls and let the syscall package deal with platform specific issues (as some packages may call syscall directly). In this case, I think syscall_libc.go should work.

Reply:
That's true. But Nintendo switch has read, seek, close, open calls. Should I implement then on `syscall_libc` mapping to libc native? That would break `picolibc` right? Maybe its better to create a `syscall_nintendoswitch`.

I will do that tonight.

Reply:
Yes, I think mapping the "syscalls" to the libc `open`, `close` etc calls is the best strategy.

> That would break `picolibc` right?

I'm not sure what you mean. You can only use one libc. So if you're using libnx (which seems to act as a libc), you shouldn't use picolibc. Picolibc is only used if you say so in the target JSON file.

Reply:
I mean break the picolibc for non nintendo switch cases. I will check if picolibc implements open / read / close calls as well.

Reply:
> I mean break the picolibc for non nintendo switch cases.

I'm not sure what you mean here. TinyGo only uses picolibc on baremetal systems, where the libc version of the syscall package is not used. In any case, this PR shouldn't modify non-Switch targets.

Reply:
Ahh so its okay. I will implement the other methods and keep in the default libc syscall.

Reply:
Done, everything on syscall_libc now.